File: 003-download-image.py
import requests
from PIL import Image

url = "https://source.unsplash.com/random"
response = requests.get(url)

with open("original.png", "wb") as file:
    file.write(response.content)

# Load the image
image = Image.open("original.png")

# Resize the image 50% smaller
resized_image = image.resize((int(image.width * 0.5), int(image.height * 0.5)))

# Convert the image to greyscale
greyscale_image = resized_image.convert("L")

# Save the greyscale image
greyscale_image.save("greyscale.png")

File: 003-download-image.js
const axios = require('axios');
const fs = require('fs').promises;
const sharp = require('sharp');

async function downloadAndProcessImage() {
    const url = 'https://source.unsplash.com/random';
    const response = await axios.get(url, { responseType: 'arraybuffer' });
    const originalImage = Buffer.from(response.data, 'binary');

    await fs.writeFile('original.png', originalImage);

    const image = sharp(originalImage);
    const metadata = await image.metadata();

    const resizedImage = await image
        .resize({ width: Math.round(metadata.width * 0.5) })
        .toBuffer();

    await sharp(resizedImage)
        .greyscale()
        .toFile('greyscale.png');
}

downloadAndProcessImage().catch(console.error);